cmd/diff.go

```diff
@@ -6,36 +6,73 @@ import (
 
 	"github.com/bmatcuk/doublestar"
 	"github.com/charypar/monobuild/diff"
+	"github.com/charypar/monobuild/graph"
+	"github.com/charypar/monobuild/manifests"
 	"github.com/spf13/cobra"
 )
 
+var baseBranch string
+var mainBranch bool
+var dotHighlight bool
+
 var diffCmd = &cobra.Command{
 	Use:   "diff",
-	Short: "List changed components",
-	Long:  `List changed components based on git history and dependency graph`,
-	Run:   diffFn,
-}
+	Short: "Build schedule for components affected by git changes",
+	Long: `Create a build schedule based on git history and dependency graph.
+Each line in the output is a component and its dependencies. 
+The format of each line is:
 
-var baseBranch string
-var mainBranch bool
+<component>: <dependency>, <dependency>, <dependency>, ...
+
+Diff can output either the build schedule (using only strong dependencies) or 
+the original dependeny graph (using all dependencies).`,
+	Run: diffFn,
+}
 
 func init() {
 	rootCmd.AddCommand(diffCmd)
 
 	diffCmd.Flags().StringVar(&baseBranch, "base-branch", "master", "Base branch to use for comparison")
 	diffCmd.Flags().BoolVar(&mainBranch, "main-branch", false, "Run in main branch mode (i.e. only compare with parent commit)")
+	diffCmd.Flags().BoolVar(&printDependencies, "dependencies", false, "Ouput the dependencies, not the build schedule")
+	diffCmd.Flags().BoolVar(&dotFormat, "dot", false, "Print in DOT format for GraphViz")
+}
+
+func joinErrors(message string, errors []error) error {
+	errstrings := make([]string, len(errors))
+	for i, e := range errors {
+		errstrings[i] = string(e.Error())
+	}
+
+	return fmt.Errorf("%s\n%s", message, strings.Join(errstrings, "\n"))
 }
 
 func diffFn(cmd *cobra.Command, args []string) {
```

Comment:
It would be useful to pull this function out into another package, so that `monobuild` can be used as a library in another application.

Reply:
i.e. escape from `/cmd` (and `cobra`) as soon as possible.